cmd/neutrond/config.go

```diff
@@ -0,0 +1,239 @@
+	}
+	return cmd
+}
+
+func runConfigCmd(cmd *cobra.Command, args []string) error {
+	clientCtx := client.GetClientContextFromCmd(cmd)
+	configPath := filepath.Join(clientCtx.HomeDir, "config")
+
+	conf, err := getClientConfig(configPath, clientCtx.Viper)
+	if err != nil {
+		return fmt.Errorf("couldn't get client config: %v", err)
+	}
+
+	ncc := NeutronCustomClient{
+		*conf,
+		os.Getenv("NEUTROND_GAS"),
+		os.Getenv("NEUTROND_GAS_PRICES"),
+		os.Getenv("NEUTROND_GAS_ADJUSTMENT"),
+
+		os.Getenv("NEUTROND_FEES"),
+		// os.Getenv("NEUTROND_FEE_GRANTER"),
+		// os.Getenv("NEUTROND_FEE_PAYER"),
+
+		os.Getenv("NEUTROND_NOTE"),
+	}
+
+	switch len(args) {
+	case 0:
+		s, err := json.MarshalIndent(ncc, "", "\t")
+		if err != nil {
+			return err
+		}
+
+		cmd.Println(string(s))
+
+	case 1:
+		// it's a get
+		key := args[0]
+
+		switch key {
+		case flags.FlagChainID:
+			cmd.Println(conf.ChainID)
+		case flags.FlagKeyringBackend:
+			cmd.Println(conf.KeyringBackend)
+		case tmcli.OutputFlag:
+			cmd.Println(conf.Output)
+		case flags.FlagNode:
+			cmd.Println(conf.Node)
+		case flags.FlagBroadcastMode:
+			cmd.Println(conf.BroadcastMode)
+
+		// Custom flags
+		case flags.FlagGas:
+			cmd.Println(ncc.Gas)
+		case flags.FlagGasPrices:
+			cmd.Println(ncc.GasPrices)
+		case flags.FlagGasAdjustment:
+			cmd.Println(ncc.GasAdjustment)
+		case flags.FlagFees:
+			cmd.Println(ncc.Fees)
+		// TODO: add with newer version of cosmos-sdk
+		// case flags.FlagFeeGranter:
+		//	cmd.Println(ncc.FeeGranter)
+		// case flags.FlagFeePayer:
+		//	cmd.Println(ncc.FeePayer)
+		case flags.FlagNote:
+			cmd.Println(ncc.Note)
+		default:
+			err := errUnknownConfigKey(key)
+			return fmt.Errorf("couldn't get the value for the key: %v, error:  %v", key, err)
+		}
+
+	case 2:
+		// it's set
+		key, value := args[0], args[1]
+
+		switch key {
+		case flags.FlagChainID:
+			ncc.ChainID = value
+		case flags.FlagKeyringBackend:
+			ncc.KeyringBackend = value
+		case tmcli.OutputFlag:
+			ncc.Output = value
+		case flags.FlagNode:
+			ncc.Node = value
+		case flags.FlagBroadcastMode:
+			ncc.BroadcastMode = value
+		case flags.FlagGas:
+			ncc.Gas = value
+		case flags.FlagGasPrices:
+			ncc.GasPrices = value
+			ncc.Fees = "" // resets since we can only use 1 at a time
+		case flags.FlagGasAdjustment:
+			ncc.GasAdjustment = value
+		case flags.FlagFees:
+			ncc.Fees = value
+			ncc.GasPrices = "" // resets since we can only use 1 at a time
+			// TODO: add with newer version of cosmos-sdk
+		// case flags.FlagFeeGranter:
+		//	 ncc.FeeGranter = value
+		// case flags.FlagFeePayer:
+		//	 ncc.FeePayer = value
+		case flags.FlagNote:
+			ncc.Note = value
+		default:
+			return errUnknownConfigKey(key)
+		}
+
+		confFile := filepath.Join(configPath, "client.toml")
+		if err := writeConfigToFile(confFile, &ncc); err != nil {
+			return fmt.Errorf("could not write client config to the file: %v", err)
+		}
+
+	default:
+		panic("could not execute config command")
```

Comment:
as a user I'd like to see the reason for it which seems to be too many arguments: 2 expected